lib/validate.rb

```diff
@@ -0,0 +1,70 @@
+class Validate
+
+    attr_accessor :message
+    def initialize
+        @message = {"error": {}}
+    end
+
+    def validate_startgame(game_id, datastore)
+        if datastore.has_key?(game_id)
+            @message[:error].store("game", "game #{game_id} already exists")
+            return false
+        end
+        return true
+    end
+
+    def validate_available_moves(game_id, datastore)
+        if !datastore.has_key?(game_id)
+            @message[:error].store("game", "game #{game_id} hasn't yet started")
+            return false
+        else
+            return true
+        end
+    end
+
+    def validate_turns(game_id, datastore)
+        if !datastore.has_key?(game_id)
+            @message[:error].store("turn", "game #{game_id} not started")
+            return false
+        else
+            return true
+        end
+    end
+
+    def validate_play(payload, datastore)
+        if !payload['game_id']
+            @message[:error].store("game_id", "game id must be present")
+        elsif !payload['player']
+            @message[:error].store("player", "player must be present")
+        elsif !payload['position']
+            @message[:error].store("position", "position must be present")
+        elsif !datastore.has_key?(payload['game_id'])
+            @message[:error].store('game', "game hasn't yet started")
+        elsif !validate_turn(payload, datastore)
+            @message[:error].store('turn', "wrong turn")
+        elsif !validate_position(payload, datastore)
+            @message[:error].store('position', "already occupied")
+        else
+            return true
+        end
```

Comment:
I'm not in love with the multiple `elsif` here. Can you think of a way to improve this?